compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -292,7 +292,15 @@ impl<'a> Parser<'a> {
                 let op_span = self.prev_token.span.to(self.token.span);
                 // Eat the second `+`
                 self.bump();
-                lhs = self.recover_from_postfix_increment(lhs, op_span)?;
+                let prev_is_semi = {
+                    if let Ok(prev_code) = self.sess.source_map().span_to_prev_source(lhs.span) &&
+                          prev_code.trim_end().ends_with(";") {
+                            true
+                    } else {
+                        false
+                    }
+                };
```

Comment:
This is a bit of a hack, but fair given where it lives. I would like to do a perf run of an alternative approach: modify `lhs` to hold a `starts_statement: bool` field. I think that it might hit memory a bit, but that would be far less brittle to refactors breaking this heuristic. Would you have time to try doing that as part of this PR?

Reply:
You means add a field with boolean type to `LhsExpr` type?

Reply:
Updated the code.
I don't think there is a perf regerssion here, we are at error report codepath,
but you may have a double-check.

Reply:
Yes, maybe cost a little bit more memory.